provider/pkg/await/error.go

```diff
@@ -38,6 +45,22 @@ func (ce *cancellationError) Object() *unstructured.Unstructured {
 	return ce.object
 }
 
+// namespaceError represents an operation that failed because the namespace didn't exist.
+type namespaceError struct {
+	object *unstructured.Unstructured
+}
+
+var _ error = (*namespaceError)(nil)
+var _ PreviewError = (*namespaceError)(nil)
```

Comment:
I'm not familiar with this approach - what does it do? Is it to ensure that those types implement the contract correctly?

Reply:
Yes, this would cause a compile-time error if the type didn't implement the interface.